src/libraries/Microsoft.PowerFx.Interpreter/Marshal/ObjectMarshaler.cs

```diff
@@ -0,0 +1,73 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.Public.Types;
+using Microsoft.PowerFx.Core.Public.Values;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Marshal a specific type of object to a record. 
+    /// </summary>
+    [DebuggerDisplay("ObjMarshal({Type})")]
+    public class ObjectMarshaler : ITypeMarshaller
+    {
+        // Map fx field name to a function produces the formula value given the dotnet object.
+        private readonly IReadOnlyDictionary<string, Func<object, FormulaValue>> _mapping;
+
+        /// <inheritdoc/>
+        public FormulaType Type { get; private set; }
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="ObjectMarshaler"/> class.
+        /// </summary>
+        /// <param name="type">The FormulaType that these objects product.</param>
+        /// <param name="fieldMap">A mapping of fx field names to functions that produce that field. </param>
+        public ObjectMarshaler(FormulaType type, IReadOnlyDictionary<string, Func<object, FormulaValue>> fieldMap)
+        {
+            if (!(type is RecordType))
```

Comment:
Not even a nit, no need to change this, but you can do `type is not RecordType` now.